File: research/acceleration_lab/ipm_accellab_plot_ABC_inner.py
"""Standalone scientific figure from saved curves; no numerical field recomputation."""
import json
import sys
from pathlib import Path
import matplotlib
matplotlib.use("Agg")
import matplotlib.pyplot as plt
import numpy as np

root=Path(sys.argv[1]).resolve()
report=json.loads((root/"report.json").read_text())
plt.rcParams.update({"font.size":10,"axes.spines.top":False,"axes.spines.right":False,"pdf.fonttype":42})
fig,axes=plt.subplots(2,2,figsize=(10.5,7),layout="constrained")
colors=["#2166ac","#b2182b","#238b45"]
curves={}
for k,c in enumerate(report["cases"]):
    for direction in (1,2):
        z=np.loadtxt(root/f'{c["label"]}_{direction}_curve.csv',delimiter=',');curves[k,direction]=z
        label=f'{c["label"][0]}: H{[8,16,8][k]}, target {[32,32,42][k]}, tau={c["canonicalTime"]:.4f}'
        axes[0,direction-1].plot(z[:,0],z[:,1],color=colors[k],lw=1.8,label=label)
        axes[0,direction-1].plot(z[:,0],z[:,2],color=colors[k],lw=.8,ls='--',alpha=.7)
for j,(a,b) in enumerate([(0,1),(0,2),(1,2)]):
    for direction in (1,2):
        aa,bb=curves[a,direction],curves[b,direction]
        assert np.array_equal(aa[:,0],bb[:,0])
        ax=axes[1,direction-1]
        ax.plot(aa[:,0],bb[:,1]-aa[:,1],color=colors[j],lw=1.5,label=f'{"ABC"[b]} - {"ABC"[a]}')
        ax.plot(aa[:,0],bb[:,2]-aa[:,2],color=colors[j],lw=.7,ls='--',alpha=.65)
for direction,title in enumerate(["Wall: x = a + wall-width * xi, y = 0", "Vertical: x = a, y = vertical-width * eta"]):
    axes[0,direction].set_title(title)
    axes[0,direction].set_ylabel("rho_x / continuous peak")
    axes[1,direction].set_ylabel("Aligned shape difference")
    axes[1,direction].axhline(0,color=".5",lw=.6)
    for row in (0,1):
        axes[row,direction].set_xlabel("xi" if direction==0 else "eta")
        axes[row,direction].grid(alpha=.2)
    axes[0,direction].legend(fontsize=8,loc="best")
    axes[1,direction].legend(fontsize=9)
fig.suptitle(f'Actual common physical time t = {report["samePhysicalEndpoint"]:.12f}\nOwn C1 peak + connected 90% widths; solid Hermite / dashed linear',fontsize=13)
fig.savefig(root/"ABC_aligned_inner_shapes.png",dpi=180)
fig.savefig(root/"ABC_aligned_inner_shapes.pdf")
print(root/"ABC_aligned_inner_shapes.png")

File: research/acceleration_lab/ipm_accellab_plot_ABC_inner.m
function ipm_accellab_plot_ABC_inner(out)
%IPM_ACCELLAB_PLOT_ABC_INNER Plot saved profiles without field recomputation.
d=load(fullfile(out,'profiles.mat'),'report','curves');r=d.report;c=d.curves;
f=figure('Visible','off','Color','w','Position',[0,0,1200,800]);cleanup=onCleanup(@()close(f));
layout=tiledlayout(f,2,2,'TileSpacing','compact','Padding','compact');
colors=[.13,.4,.67;.7,.094,.17;.137,.545,.27];
for direction=1:2
    ax=nexttile(layout,direction);hold(ax,'on');handles=gobjects(1,3);labels=cell(1,3);
    for k=1:3
        z=c{k,direction};handles(k)=plot(ax,z.coordinate,z.hermite,'Color',colors(k,:),'LineWidth',1.8);
        plot(ax,z.coordinate,z.linear,'Color',colors(k,:),'LineStyle','--','LineWidth',.6);
        labels{k}=sprintf('%s, tau=%.4f',r.cases(k).label,r.cases(k).canonicalTime);
    end
    ylabel(ax,'rho_x / continuous peak');legend(ax,handles,labels,'Interpreter','none','Location','best','FontSize',9);
    if direction==1,title(ax,'Wall: x = a + w_x xi, y = 0');xlabel(ax,'xi');
    else,title(ax,'Vertical: x = a, y = w_y eta');xlabel(ax,'eta');end
    grid(ax,'on');box(ax,'off');
    ax=nexttile(layout,direction+2);hold(ax,'on');pairs=[1,2;1,3;2,3];
    for k=1:3
        a=c{pairs(k,1),direction};b=c{pairs(k,2),direction};assert(isequal(a.coordinate,b.coordinate));
        handles(k)=plot(ax,a.coordinate,b.hermite-a.hermite,'Color',colors(k,:),'LineWidth',1.5);
        plot(ax,a.coordinate,b.linear-a.linear,'Color',colors(k,:),'LineStyle','--','LineWidth',.6);
    end
    yline(ax,0,'Color',[.5,.5,.5],'LineWidth',.5);legend(ax,handles,{'B - A','C - A','C - B'},'Location','best');
    ylabel(ax,'Aligned shape difference');grid(ax,'on');box(ax,'off');
    if direction==1,xlabel(ax,'xi');else,xlabel(ax,'eta');end
end
title(layout,sprintf(['Actual common physical time t = %.12f\n', ...
    'Own C1 peak + connected 90%% widths; solid Hermite / dashed linear'],r.samePhysicalEndpoint),'FontSize',14);
exportgraphics(f,fullfile(out,'ABC_aligned_inner_shapes.png'),'Resolution',180);
exportgraphics(f,fullfile(out,'ABC_aligned_inner_shapes.pdf'),'ContentType','vector');
end
